rental.rb

```diff
@@ -1,13 +1,21 @@
 class Rental
-  attr_accessor :date, :book, :person
+  attr_reader :book, :person
+  attr_accessor :date
 
-  def initialize(date, person, book)
+  def initialize(date, book, person)
     @date = date
-
+    @book = book
     @person = person
-    person.rental << self
+    book.rentals << self
+    person.rentals << self
+  end
 
-    @book = book
-    book.rental << self
+  def to_json(*args)
+    {
+      JSON.create_id => self.class.name,
+      'date' => @date,
```

Comment:
- [ ] Entering a date option for creating rental is yielding an error. Kindly ensure to improve upon this aspect.

**Error:**
![image](https://user-images.githubusercontent.com/12745474/144580850-6cae499f-a41b-4077-8fe5-6e174ab2bfda.png)

Reply:
This was not part of the requirements.